src/backend/cdb/dispatcher/cdbdisp.c

```diff
@@ -520,6 +521,19 @@ cleanup_dispatcher_handle(dispatcher_handle_t *h)
 	cdbdisp_destroyDispatcherState(h->dispatcherState);
 }
 
+static void
+destroy_dispatcher_state_and_handle(dispatcher_handle_t *h)
+{
+	if (h->dispatcherState == NULL)
+	{
+		destroy_dispatcher_handle(h);
+		return;
+	}
+
+	h->dispatcherState->forceDestroyGang = true;
+	cdbdisp_destroyDispatcherState(h->dispatcherState);
+}
+
```

Comment:
Is it possible to combine similar non-public functions `cleanup_dispatcher_handle` and `destroy_dispatcher_state_and_handle`?